toolchain/gcc-14.2.0-13-arm-multilib.patch

```diff
@@ -0,0 +1,208 @@
+diff --git a/gcc/config.gcc b/gcc/config.gcc
+--- a/gcc/config.gcc
++++ b/gcc/config.gcc
+@@ -1475,7 +1475,7 @@ arm*-*-phoenix*)
+ 	tm_file="elfos.h arm/unknown-elf.h arm/elf.h arm/bpabi.h"
+ 	tm_file="${tm_file} newlib-stdint.h phoenix.h"
+ 	tm_file="${tm_file} arm/aout.h arm/arm.h"
+-	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-aprofile arm/t-rmprofile arm/t-multilib"
++	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-multilib arm/t-phoenix"
+ 	with_float="soft" # "auto" fpu
+ 	TM_MULTILIB_CONFIG="aprofile,rmprofile"
+ 	target_cpu_cname="arm7tdmi"
+diff --git a/gcc/config/arm/t-multilib b/gcc/config/arm/t-multilib
+index be63edd85c4..14c6c9875f1 100644
+--- a/gcc/config/arm/t-multilib
++++ b/gcc/config/arm/t-multilib
+@@ -225,4 +225,3 @@ MULTILIB_REUSE		+= $(foreach ABI, hard softfp, \
+ MULTILIB_REUSE		+= $(foreach MODE, arm thumb, \
+ 			     $(foreach ARCH, armv7 armv7-r armv8-r, \
+ 			       mthumb/march.$(ARCH)/mfloat-abi.soft=m$(MODE)/march.$(ARCH)/mfloat-abi.softfp))
+-
+diff --git a/gcc/config/arm/t-phoenix b/gcc/config/arm/t-phoenix
+--- a/gcc/config/arm/t-phoenix
++++ b/gcc/config/arm/t-phoenix
+@@ -16,16 +16,45 @@
+ # along with GCC; see the file COPYING3.  If not see
+ # <http://www.gnu.org/licenses/>.
+ 
+-MULTILIB_OPTIONS	= marm/mthumb
+-MULTILIB_DIRNAMES	= arm thumb
+-MULTILIB_EXCEPTIONS	=
+-MULTILIB_MATCHES	=
+-MULTILIB_REUSE		=
+-MULTILIB_REQUIRED	=
+-
+-MULTILIB_OPTIONS	+= mfloat-abi=hard
+-MULTILIB_DIRNAMES	+= fpu
+-MULTILIB_MATCHES	+= mfloat-abi?hard=mhard-float
+-
+-MULTILIB_OPTIONS	+= mno-thumb-interwork/mthumb-interwork
+-MULTILIB_DIRNAMES	+= normal interwork
++# Add phoenix options to reuses and requires.
++
++s-mlib: $(srcdir)/config/arm/t-phoenix
++
++# Find all RM archs.
++RM_ARCHS			:= $(subst /, ,march=armv7-r+fp.sp/$(MULTI_ARCH_OPTS_RM))
++# Create matches for requires and reuses.
++RM_MATCHES_REQ		:= $(foreach ARCH, $(RM_ARCHS), mthumb/$(ARCH)% marm/$(ARCH)%)
++RM_MATCHES_REUSE	:= $(foreach TARGET, $(RM_MATCHES_REQ), $(subst =,.,$(subst .,\.,$(RM_MATCHES_REQ))))
++
++# NOTE: Multilib options in MULTILIB_REQUIRED must be in the same order as they were added to the MULTILIB_OPTIONS variable.
++# So to easily add custom options at the end of REQUIRE we have to add them at the end of OPTIONS and DIRS.
++MULTILIB_OPTIONS	+= fPIC mno-pic-data-is-text-relative/mfdpic
++MULTILIB_DIRNAMES	+= pic no-pditr fdpic
++
++MULTILIB_MATCHES	+= fPIC=fpic fPIC=fPIE fPIC=fpie
++
++PHOENIX_OPTS		:= /fPIC/mno-pic-data-is-text-relative
++PHOENIX_OPTS		+= /fPIC/mfdpic
++
++# Finally, add Phoenix options to RM multilibs.
++# Keep the orignal requires because of kernel and plo.
++
++RM_REQUIRED			:= $(filter $(RM_MATCHES_REQ),$(MULTILIB_REQUIRED))
++
++RM_REUSE			:= $(filter $(RM_MATCHES_REUSE),$(MULTILIB_REUSE))
++
++MULTILIB_REQUIRED	+= $(foreach REQ, $(RM_REQUIRED), \
```

Comment:
nitpicking: the original t-files from gcc use tab-width 6 (you're using probably 4 in your editor - that's why the lines do not align sometimes?)

Reply:
update: not fixed even after setting tab width to 6 my editor is acting up

Reply:
update: no I think it is alright the tab width in github is just different but when set to 6 it is all good both in VIM and vscode

Reply:
ack